airflow/hooks/base.py

```diff
@@ -77,6 +78,8 @@ def get_connection(cls, conn_id: str) -> Connection:
         :param conn_id: connection id
         :return: connection
         """
+        conn_id = settings.connection_policy(conn_id=conn_id)
```

Comment:
I wonder if `Connection.get_connection_from_secrets` would be a better location

Reply:
Possibly, but I don't think it matters that much?

The policy would have to be applied before the first request to any secrets system. We could apply it inside `Connection.get_connection_from_secrets`, but it would have to be the first statement in that method, because after that we start requesting secrets. That applies the policy one level deeper in the stack but logically it wouldn't make a difference.